### ISSUE TITLE
医学图像

### ISSUE THREAD

**Issue by Saillxl** [WZH0120/SAM2-UNet#5]
感谢作者的开源代码：我运行了2分类的2D医学图像数据集，训练集大概两三百张，eval.py的结果很差：mDice:        0.168；mIoU:         0.111。请问有人知道原因吗？

**Comment by xiongxyowo** [comment]
Hi, is the dataset publicly available? If so, you can provide the dataset name so we can try to verify it. It is worth noting that you may need to adjust the loss function for segmentation tasks with num_class >= 2 (a cross entropy should be fine). Additionally, `eval.py` and `test.py` is designed for binary segmentation only. You need to use your own scripts for some customized datasets.

**Comment by Saillxl** [comment]
Thank you for your prompt reply, here is the download link：https://scholar.cu.edu.eg/?q=afahmy/pages/dataset

**Comment by xiongxyowo** [comment]
Hello, we just trained on BUSI and the results don't seem to be bad. We randomly selected 80% of benign and malignant for training and 20% for testing. Note that we did not use the normal subset for testing, as negative samples may be harmful to the metric calculation (easily 0). Maybe you could try expanding your training set.

**Comment by Saillxl** [comment]
Thanks to the author, what is the hyperparameter you set? The highest mice I tried was 0.44

**Comment by xiongxyowo** [comment]
We used the default training hyperparameter in the code.

**Comment by Saillxl** [comment]
ok, thank you for your patience！

**Comment by hczyni** [comment]
> 感谢作者的开源代码：我运行了2分类的2D医学图像数据集，训练集大概两三百张，eval.py的结果很差：mDice: 0.168；mIoU: 0.111。请问有人知道原因吗？
Dear AI Engineer, how can I calculate evaluation metrics such as Dice, IOU, and HD? Could you provide me with some code? I found some code online regarding this topic, but I encountered some issues when running it. I would greatly appreciate it if you could share your code for these metrics! Thank you very much!
